ast/compile.go

```diff
@@ -844,12 +855,102 @@ func (c *Compiler) checkSafetyRuleHeads() {
 	}
 }
 
+//CompileSchemas takes a jsonschema and compiles it.
+func CompileSchemas(byteSchema []byte, goSchema interface{}) (*gojsonschema.Schema, error) {
```

Comment:
I think we can hide this function as an implementation detail of the compilation process. Basically the compiler should accept a set of raw schemas (via `WithSchemas` above) and then when the type checking stage is executed, we can parse and compile the raw schemas, translate them into the type system, and add them into the environment.

Reply:
Done